src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -327,7 +375,88 @@ internal class DefaultRazorSemanticTokensInfoService : RazorSemanticTokensInfoSe
             return csharpResponse.Result;
         }
 
-        private SemanticRange DataToSemanticRange(int lineDelta, int charDelta, int length, int tokenType, int tokenModifiers, SemanticRange? previousSemanticRange = null)
+        private async Task<SemanticTokens?> GetMatchingCSharpEditsResponseAsync(
+            TextDocumentIdentifier textDocumentIdentifier,
+            long? documentVersion,
+            string? previousResultId,
+            IReadOnlyList<int> previousCSharpTokens,
+            CancellationToken cancellationToken)
+        {
+            Assumes.NotNull(previousResultId);
```

Comment:
If we assume this is NotNull shouldn't `previousResultId` be `string` instead of `string?`?